src/ledger/LedgerTxnAccountSQL.cpp

```diff
@@ -367,6 +370,348 @@ LedgerTxnRoot::Impl::deleteAccount(LedgerKey const& key)
+            inflationDestInds.push_back(soci::i_ok);
+        }
+        else
+        {
+            inflationDests.push_back("");
+            inflationDestInds.push_back(soci::i_null);
+        }
+        flags.push_back(account.flags);
+        homeDomains.push_back(account.homeDomain);
+        thresholds.push_back(decoder::encode_b64(account.thresholds));
+        lastModifieds.push_back(e.lastModifiedLedgerSeq);
+        assert(account.signers.empty());
+        if (account.ext.v() >= 1)
+        {
+            buyingLiabilities.push_back(account.ext.v1().liabilities.buying);
+            sellingLiabilities.push_back(account.ext.v1().liabilities.selling);
+            liabilitiesInds.push_back(soci::i_ok);
+        }
+        else
+        {
+            buyingLiabilities.push_back(0);
+            sellingLiabilities.push_back(0);
+            liabilitiesInds.push_back(soci::i_null);
+        }
+    }
+
+    // NB: at the moment we only have two flavors of database support, this
+    // condition 2-way split will need to change if we support more.
+    if (mDatabase.isSqlite())
+    {
+        sociGenericBulkUpsertAccounts(
+            mDatabase, accountIDs, balances, seqNums, subEntryNums,
+            inflationDests, inflationDestInds, flags, homeDomains, thresholds,
+            lastModifieds, buyingLiabilities, sellingLiabilities,
+            liabilitiesInds);
+    }
+    else
+    {
+        postgresSpecificBulkUpsertAccounts(
+            mDatabase, accountIDs, balances, seqNums, subEntryNums,
+            inflationDests, inflationDestInds, flags, homeDomains, thresholds,
+            lastModifieds, buyingLiabilities, sellingLiabilities,
+            liabilitiesInds);
+    }
+}
+
+static void
+postgresSpecificBulkDeleteAccounts(Database& DB,
+                                   std::vector<std::string> const& accountIDs)
+{
+    soci::session& session = DB.getSession();
+    auto pg =
+        dynamic_cast<soci::postgresql_session_backend*>(session.get_backend());
+    PGconn* conn = pg->conn_;
+
+    std::string accountIDstr = marshalToPGArray(conn, accountIDs);
+
+    std::string sql =
+        "WITH r AS (SELECT unnest(:ids::TEXT[])) "
+        "DELETE FROM accounts WHERE accountid IN (SELECT * FROM r)";
+    auto prep = DB.getPreparedStatement(sql);
+    soci::statement& st = prep.statement();
+    st.exchange(soci::use(accountIDstr, "ids"));
+    st.define_and_bind();
+    {
+        auto timer = DB.getDeleteTimer("account");
+        st.execute(true);
+    }
+}
+
+static void
+sociGenericBulkDeleteAccounts(Database& DB,
+                              std::vector<std::string> const& accountIDs)
+{
+    std::string sql = "DELETE FROM accounts WHERE accountid = :id";
+    auto prep = DB.getPreparedStatement(sql);
+    soci::statement& st = prep.statement();
+    st.exchange(soci::use(accountIDs, "id"));
+    st.define_and_bind();
+    {
+        auto timer = DB.getUpsertTimer("account");
```

Comment:
I think this should be `getDeleteTimer`.

Reply:
Yeah, copypasta, fixed.